MAVProxy/modules/mavproxy_misseditor/missionEditorFrame.py

```diff
@@ -673,6 +689,40 @@ def fix_jumps(self, row_selected, delta):
                 if p1 > row_selected and p1+delta>0:
                     self.grid_mission.SetCellValue(row, ME_P1_COL, str(float(p1+delta)))
 
+    def set_grad_dist(self):
+	'''fix up distance and gradient when changing cell values'''
+	numrows = self.grid_mission.GetNumberRows()
+	for row in range(1,numrows):
+	    command = self.grid_mission.GetCellValue(row, ME_COMMAND_COL)
+	    command_prev = self.grid_mission.GetCellValue(row - 1, ME_COMMAND_COL)
+            lat = float(self.grid_mission.GetCellValue(row, ME_LAT_COL))
+            lon = float(self.grid_mission.GetCellValue(row, ME_LON_COL))
+            if (lat == 0) and (lon == 0):
+                dist = 0.0
+                grad = 0.0
+	    else :
+              if "NAV" in command:
+                row_prev = row - 1
+		while "NAV" not in command_prev:
+                    command_prev = self.grid_mission.GetCellValue(row_prev - 1, ME_COMMAND_COL)
+                    row_prev = row_prev - 1
+		prev_lat = float(self.grid_mission.GetCellValue(row_prev, ME_LAT_COL))
+		prev_lon = float(self.grid_mission.GetCellValue(row_prev, ME_LON_COL))
+		prev_alt = float(self.grid_mission.GetCellValue(row_prev, ME_ALT_COL))
+                while (prev_lat == 0) and (prev_lon == 0):
```

Comment:
check for row_prev zero here too (don't want to end up negative)

Reply:
done